lib/manageiq/rpm_build/nightly_build_purger.rb

```diff
@@ -0,0 +1,53 @@
+module ManageIQ
+  module RPMBuild
+    class NightlyBuildPurger
+      include S3Common
+
+      def run
+        candidates = {}
+        keepers    = {}
+
+        client.list_objects(:bucket => OPTIONS.rpm_repository.s3_api.bucket, :prefix => File.join("builds", "manageiq-nightly")).flat_map(&:contents).each do |object|
+          package, timestamp    = package_timestamp_from_key(object.key)
+          candidates[package] ||= []
+          keepers[package]    ||= []
+
+          if recent?(timestamp)
+            keepers[package] << object.key
+            next
+          end
+
+          position = (timestamp.to_i - 20_000_000_000_000) / 10_000 # YYMMDDHH i.e. 22091300
+          candidates[package][position] = object.key
+        end
+
+        candidates.each_key do |package|
```

Comment:
Is there some implicit sorting here? Perhaps from s3?  How do you know the last 14 are the keepers?

Reply:
So you have an array with 22110112 entries?  I'm so confused - you might have to show me.

Reply:
Yes, lots of `nil`s, but it's simple and ordered.

Reply:
that's weird, but for now it's fine - we can clean it up later.